pandas/tests/indexes/categorical/test_reindex.py

```diff
@@ -59,3 +59,10 @@ def test_reindex_missing_category(self):
         msg = "'fill_value=-1' is not present in this Categorical's categories"
         with pytest.raises(TypeError, match=msg):
             ser.reindex([1, 2, 3, 4, 5], fill_value=-1)
+
+    def test_reindex_not_category(self):
```

Comment:
can you add all 4 cases from the OP (3 working, 1 non).

Reply:
done